packages/commerce-sdk-react/src/auth/index.ts

```diff
@@ -532,7 +543,26 @@ class Auth {
                 this.logger.info(`Refresh login error. Attempting to log user in as new guest.`)
             }
         }
-        return this.loginGuestUser()
+
+        // refresh flow for TAOB
+        if (accessToken && this.isTokenExpired(accessToken)) {
+            const {isGuest, usid, loginId, isAgent} = this.parseSlasJWT(accessToken)
+            if (isAgent) {
+                return await this.queueRequest(() => this.refreshTrustedAgent(loginId, usid), isGuest)
+            }
+        }
+
+        // if a TAOB left a usid and it tries to
+        // use it, we will be stuck in a fail loop
```

Comment:
why would a `usid` become a problem here? which exact API call will fail and why is fail loop created? It would be nice to have more details for this comment

Reply:
if a `usid` is used in a TAOB auth flow, then we log out (which also logs us into guest user but leaves the `usid`), then when it tries to use the existing `usid` in the new guest token endpoint call, it throws an error.